### ISSUE TITLE
[Bug] Client authentication malformed errors

### ISSUE THREAD

**Issue by alicerunsonfedora** [hyperspacedev/hyperspace#8]
In some cases and on certain installations of Hyperspace, the authentication process fails, reporting the following error:

> Client authentication failed due to unknown client, no client authentication included, or unsupported authentication method.

This has been noticed with [gopherdonapp/app](https://github.com/gopherdonapp/app) (though this might be a different issue) and also with Hyperspace 1.0.0beta1u1 by [@nvsr@mstdn.jp](https://mstdn.jp/users/nvsr/statuses/101955149196194642) at the hyperspaceapp-next URL.

## Possible causes

* Dynamic redirect URL methods in deployment don't work correctly in Megalodon.
* HTTPS authentication conflicts

## Possible solutions
- [x] Add a parameter in `config.json` that explicitly states the URL to which Hyperspace will be hosted.
- [x] Reserve dynamic redirects for the desktop app and/or as a fallback to the parameter.
- [ ] Use hyperspaceapp:// protocol for desktop app redirection

**Comment by alicerunsonfedora** [comment]
Given that Beta 2 hasn't actually resolved said issues, I'm going to try implementing the old method as a backup plan.

**Comment by alicerunsonfedora** [comment]
Rescue login mode should be applied in Update 1 to Beta 2 this week.

![Screen Shot 2019-04-22 at 19 04 18](https://user-images.githubusercontent.com/13445064/56583596-efe1b080-65a7-11e9-904a-ae7c7374e022.png)

**Comment by GidiKroon** [comment]
I found with monitoring network traffic, there's first a POST to /api/v1/app to create the app, then a GET and POST to /oauth/authorize to get the authorization. However, I only today noticed that the first goes to mastodon.social, while the others go to the proper instance.

Turns out that in the megaladon code of createApp, the baseUrl is optional and defaults to mastodon.social. The file login module here doesn't provide ", baseurl" as third parameter to createApp, so the app is not registered on the actual instance, so that causes unauthorized issues if your instance is not mastodon.social.
